diff --git a/ios/DataSource.swift b/ios/DataSource.swift
--- a/ios/DataSource.swift
+++ b/ios/DataSource.swift
@@ -24,7 +24,7 @@ enum DataSource {
     return .bundle(resource: name, extension: ext.isEmpty ? nil : ext)
   }
 
-  static func bytes(from buffer: ArrayBufferHolder) -> DataSource {
+  static func bytes(from buffer: ArrayBuffer) -> DataSource {
     return .bytes(data: buffer.toData(copyIfNeeded: false))
   }
 
diff --git a/ios/HybridRiveFileFactory.swift b/ios/HybridRiveFileFactory.swift
--- a/ios/HybridRiveFileFactory.swift
+++ b/ios/HybridRiveFileFactory.swift
@@ -145,7 +145,7 @@ final class HybridRiveFileFactory: HybridRiveFileFactorySpec, @unchecked Sendabl
     return try fromResource(resource: resource, loadCdn: loadCdn, referencedAssets: nil)
   }
 
-  func fromBytes(bytes: ArrayBufferHolder, loadCdn: Bool, referencedAssets: ReferencedAssetsType?)
+  func fromBytes(bytes: ArrayBuffer, loadCdn: Bool, referencedAssets: ReferencedAssetsType?)
     throws -> Promise<
       (any HybridRiveFileSpec)
     > {
diff --git a/ios/HybridRiveImageFactory.swift b/ios/HybridRiveImageFactory.swift
--- a/ios/HybridRiveImageFactory.swift
+++ b/ios/HybridRiveImageFactory.swift
@@ -20,7 +20,7 @@ final class HybridRiveImageFactory: HybridRiveImageFactorySpec {
     return loadFromDataSource(.bundle(nameWithExtension: resource))
   }
 
-  func loadFromBytesAsync(bytes: ArrayBufferHolder) throws -> Promise<(any HybridRiveImageSpec)> {
+  func loadFromBytesAsync(bytes: ArrayBuffer) throws -> Promise<(any HybridRiveImageSpec)> {
     return loadFromDataSource(.bytes(from: bytes))
   }
 }
