diff --git a/node_modules/react-native-wgpu/android/src/main/java/com/webgpu/WebGPUViewManager.java b/node_modules/react-native-wgpu/android/src/main/java/com/webgpu/WebGPUViewManager.java
index 0000001..0000002 100644
--- a/node_modules/react-native-wgpu/android/src/main/java/com/webgpu/WebGPUViewManager.java
+++ b/node_modules/react-native-wgpu/android/src/main/java/com/webgpu/WebGPUViewManager.java
@@ -10,7 +10,7 @@ import com.facebook.react.uimanager.annotations.ReactProp;
 @ReactModule(name = WebGPUViewManager.NAME)
 public class WebGPUViewManager extends WebGPUViewManagerSpec<WebGPUView> {

-  public static final String NAME = "WebGPUView";
+  public static final String NAME = "WGPUWebGPUView";

   @NonNull
   @Override
diff --git a/node_modules/react-native-wgpu/android/src/newarch/com/webgpu/WebGPUViewManagerSpec.java b/node_modules/react-native-wgpu/android/src/newarch/com/webgpu/WebGPUViewManagerSpec.java
index 0000001..0000002 100644
--- a/node_modules/react-native-wgpu/android/src/newarch/com/webgpu/WebGPUViewManagerSpec.java
+++ b/node_modules/react-native-wgpu/android/src/newarch/com/webgpu/WebGPUViewManagerSpec.java
@@ -6,14 +6,14 @@ import androidx.annotation.Nullable;

 import com.facebook.react.uimanager.SimpleViewManager;
 import com.facebook.react.uimanager.ViewManagerDelegate;
-import com.facebook.react.viewmanagers.WebGPUViewManagerDelegate;
-import com.facebook.react.viewmanagers.WebGPUViewManagerInterface;
+import com.facebook.react.viewmanagers.WGPUWebGPUViewManagerDelegate;
+import com.facebook.react.viewmanagers.WGPUWebGPUViewManagerInterface;

-public abstract class WebGPUViewManagerSpec<T extends View> extends SimpleViewManager<T> implements WebGPUViewManagerInterface<T> {
+public abstract class WebGPUViewManagerSpec<T extends View> extends SimpleViewManager<T> implements WGPUWebGPUViewManagerInterface<T> {
   private final ViewManagerDelegate<T> mDelegate;

   public WebGPUViewManagerSpec() {
-    mDelegate = new WebGPUViewManagerDelegate(this);
+    mDelegate = new WGPUWebGPUViewManagerDelegate(this);
   }

   @Nullable
